JeMPI_Apps/JeMPI_UI/src/pages/settings/uniqueToGR/UniqueToGR.tsx

```diff
@@ -64,9 +69,48 @@ const UniqueToGR = ({
     setRows(
       rows.map((row: { id: any }) => (row.id === newRow.id ? updatedRow : row))
     )
+    handleUpdateConfiguration(updatedRow, updatedRow.rowIndex)
     return updatedRow
   }
 
+  const handleUpdateConfiguration = (updatedRow: any, rowIndex: number) => {
+    setConfiguration(previousConfiguration => {
+      if (!previousConfiguration) return previousConfiguration
+      const updatedConfiguration = getUpdatedConfiguration(
+        updatedRow,
+        rowIndex,
+        previousConfiguration
+      )
+      localStorage.setItem(
+        'configuration',
+        JSON.stringify(updatedConfiguration)
+      )
+      setConfiguration(updatedConfiguration)
```

Comment:
@NyashaMuusha just checking if this is right, calling setConfiguration inside setConfiguration, is there a reactivity issue you're dealing with or what is this meant to do?